### DISCUSSION TITLE
Rounded Corners for specific Corners

### DISCUSSION THREAD

**Discussion by snoweuph** [yshui/picom#918] [Ideas]
I currently am ricing my `i3` with `picom` and i've got the big problem that in the Tabbed mode the Tabbed bar is seen as an extra Window and so is completely separately rounded from the windows inside of it

![grafik](https://user-images.githubusercontent.com/76737863/200928464-cab2ab78-1c97-4030-9deb-30a99f933efb.png)

I currently can Only Exclude the i3 tabbed bar from getting rounded and only.
```
# Exclude conditions for rounded corners.
rounded-corners-exclude = [
  "class_g = 'i3-frame'",
];
```
overall like this ending like this
![grafik](https://user-images.githubusercontent.com/76737863/200928831-ff4dc8c3-277f-4fb1-ab3e-ca8f74b7ceb2.png)
I **can't** only select windows who are in a tabbed i3 environment, because there's nothing specificly differentiating them from normal windows when I look them up with `xprop`

so my request put short is:

- let me only exclude specific corners for specific windows, so I can make that bar rounded on the top and not rounded on the bottom
- some way to use  new setting on the i3 tabbed container *maybe there is one already and i'm just to stupid*

### Suggestion
The option could look like this
```
rounded-corners-exclude = [
  "window_type = 'unknown'",
  "class_g = 'i3-frame'" 0 0 1 1,
]
```

the extra 4 numbers after the selector are for the corners in clockwise order starting from top left, 0 meaning don't select these for the excluding, 1 meaning this should get excluded, so that this would end up with rounded corners on the top but none one the bottom

**Comment by snoweuph** [comment]
Well found out that there's a [PR](https://github.com/ibhagwan/picom/pull/5) for this on the picom fork of ibhagwan

**Reply by snoweuph** [reply]
Only Problem its 2 years old and only Includes Individual Corners not Individual Corners on Individual Windows

**Comment by BuddhiLW** [comment]
Any updates? Would be nice to know :)

**Reply by snoweuph** [reply]
No, if you have interest in this, just put an upvote on the original post. Thx

Also X is dying, so I suggest trying out Hyprland and other Wayland based compositors and Window Managers.

I personally moved on to KDE Plasma with a lot of Tweaks and Khröhnkite

**Comment by carlosfmesilva** [comment]
@yshui do you have an example on how to round top corners using the windows shader?
